### PR TITLE
Update version of backup-container image

### PR DESCRIPTION
**Pull request by grdryn**
JIRA: https://issues.redhat.com/browse/INTLY-10328

This should be covered during upgrade by the "Upgrade product images"
task, although I'm not 100% sure how the `upgrade_product_roles`
variable gets populated.

Upgrade log: https://gist.github.com/grdryn/2a44f5f931e9731be3cf991645165376#file-intly-10328-upgrade-log-L76

Images referenced in cronjobs after upgrade, showing all at the expected version:
```
# oc describe cronjobs --selector=monitoring-key=middleware --all-namespaces | grep Image
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
    Image:      quay.io/integreatly/backup-container:1.0.16
```

## Verification Steps
As the verifier of the PR the following process should be done:

### Installation Verification
- Ensure the author of the PR has attached a log of the installation run from his branch to the jira or pr and check that it exited as expected.
- Verify the fresh installation is correct on cluster provided by PR author
### Upgrade Verification
- After installation verification, notify the PR author to begin an upgrade on their cluster
- Ensure the developer of the PR has attached a log of the upgrade run from his branch to the jira or pr and check that it exited as expected.
- If possible, look at the tasks that ran and see they match the PR
- Verify the upgrade is correct on cluster provided by PR author

## Checklist

- [x] Updated [Changelog](https://github.com/integr8ly/installation/blob/master/CHANGELOG.md)
- [ ] Tested Installation
- [ ] Tested Uninstallation
- [ ] Tested or Created follow on for Upgrade

Is an upgrade task required and are there additional steps needed to test this?
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by jessesarn** [comment]
@grdryn I believe you need to add **backup** to the _upgrade_product_roles_ here: https://github.com/integr8ly/installation/blob/master/playbooks/group_vars/all/upgrade.yml#L3

This should cause the upgrade_images to be called in the backup role as part of this list iteration to cover upgrades: https://github.com/integr8ly/installation/blob/master/playbooks/upgrade.yml#L29-L33

Fresh install is covered with what you already have.

**Comment by grdryn** [comment]
@jessesarn hmm, I saw that alright, but I was thinking that I probably needed to set it somewhere else, based on the `# DO NOT EDIT This file is auto generated!!` comment on the top of that file. Is that comment correct, or should I be manually updating it? :thinking:

**Comment by jessesarn** [comment]
Sorry, you're right.  The scripts/release.sh should be able to populate this come release time.

However, you can test the update images process by adding backup to the roles list var locally.  It doesn't need to be changed for this PR itself.

**Comment by matskiv** [comment]
As far as I can tell, `playbooks/group_vars/all/upgrade.yml` is reset when final release tag is cut, and it doesn't edit it before that.
Usually delorean(EWS) is adding new items to the `upgrade_product_roles` list, but in this case we need to do it.

**Comment by grdryn** [comment]
Thanks folks, I've updated it now. I'm currently preparing an environment for upgrade and will post logs once done.

File: roles/backup/tasks/upgrade_images.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     cronjobs_names_list: "{{ cronjobs_names.stdout_lines }}"
 
 - name: Patch the image in each cronjob
-  shell: "oc patch cronjob {{ item.0 }} -n {{ backup_namespace }} --type json -p '[{\"op\":\"replace\", \"path\":\"/spec/jobTemplate/spec/template/spec/containers/0/image\", \"value\":\"{{ backup_image }}\"}]'"
+  shell: "oc patch cronjob {{ item }} -n {{ backup_namespace }} --type json -p '[{\"op\":\"replace\", \"path\":\"/spec/jobTemplate/spec/template/spec/containers/0/image\", \"value\":\"{{ backup_image }}\"}]'"
```
Comment (grdryn) [inline roles/backup/tasks/upgrade_images.yml:12]: See the commit message description for why this change was necessary.
